test/benchmark/main.cc

```diff
@@ -21,6 +21,19 @@ static bool skip_expensive_benchmarks = false;
 // separated by --.
 // TODO(pgenera): convert this to abseil/flags/ when benchmark also adopts abseil.
 int main(int argc, char** argv) {
+
+  bool is_help_flag = false;
+
+  for (int i = 1; i < argc; ++i) {
+    if (strcmp(argv[i], "--help") == 0) {
+      is_help_flag = true;
+    }
+  }
+
+  if (!is_help_flag) {
```

Comment:
Can you add a comment explaining why --help needs to be treated in a special way?

Reply:
if we don't take care of the `--help` separately, then the output is actually "benchmark --help".
```
benchmark [--benchmark_list_tests={true|false}]
          [--benchmark_filter=<regex>]
          [--benchmark_min_time=<min_time>]
          [--benchmark_repetitions=<num_repetitions>]
          [--benchmark_report_aggregates_only={true|false}]
          [--benchmark_display_aggregates_only={true|false}]
          [--benchmark_format=<console|json|csv>]
          [--benchmark_out=<filename>]
          [--benchmark_out_format=<json|console|csv>]
          [--benchmark_color={auto|true|false}]
          [--benchmark_counters_tabular={true|false}]
          [--v=<verbosity>]
```
Here, the flag details like `--skip_expensive_benchmarks` and other custom-defined flags are missing. 
That's why we are showing the older help results.

Reply:
Preferring help documentation for the custom defined flags makes sense to me.  Could you add a comment to the code that explains that we're checking for the --help flag in order to get that behavior?